mne/beamformer/_rap_music.py

```diff
@@ -264,26 +269,62 @@ def rap_music(
 
     .. versionadded:: 0.9.0
     """
-    info = evoked.info
-    data = evoked.data
-    times = evoked.times
+    return _rap_music(evoked, forward, noise_cov, n_dipoles, return_residual, False)
 
-    picks = _check_info_inv(info, forward, data_cov=None, noise_cov=noise_cov)
 
-    data = data[picks]
+@verbose
+def trap_music(
+    evoked,
+    forward,
+    noise_cov,
+    n_dipoles=5,
+    return_residual=False,
+    *,
+    verbose=None,
+):
+    """TRAP-MUSIC source localization method.
 
-    dipoles, explained_data = _apply_rap_music(
-        data, info, times, forward, noise_cov, n_dipoles, picks
-    )
+    Compute Truncated Recursively Applied and Projected MUltiple SIgnal Classification
+    (TRAP-MUSIC) on evoked data.
 
-    if return_residual:
-        residual = evoked.copy().pick([info["ch_names"][p] for p in picks])
-        residual.data -= explained_data
-        active_projs = [p for p in residual.info["projs"] if p["active"]]
-        for p in active_projs:
-            p["active"] = False
-        residual.add_proj(active_projs, remove_existing=True)
-        residual.apply_proj()
-        return dipoles, residual
-    else:
-        return dipoles
+    .. note:: The goodness of fit (GOF) of all the returned dipoles is the
+              same and corresponds to the GOF of the full set of dipoles.
+
+    Parameters
+    ----------
+    evoked : instance of Evoked
+        Evoked data to localize.
+    forward : instance of Forward
+        Forward operator.
+    noise_cov : instance of Covariance
+        The noise covariance.
+    n_dipoles : int
+        The number of dipoles to look for. The default value is 5.
+    return_residual : bool
+        If True, the residual is returned as an Evoked instance.
+    %(verbose)s
+
+    Returns
+    -------
+    dipoles : list of instance of Dipole
+        The dipole fits.
+    residual : instance of Evoked
+        The residual a.k.a. data not explained by the dipoles.
+        Only returned if return_residual is True.
+
+    See Also
+    --------
+    mne.fit_dipole
+    mne.rap_music
+
+    Notes
+    -----
+    The reference is:
+
+        N. Mäkelä, M. Stenroos, J. Sarvas, R. J. Ilmoniemi, Truncated RAP-MUSIC
+        (TRAP-MUSIC) for MEG and EEG source localization, Neuroimage, 2018.
+        doi: 10.1016/j.neuroimage.2017.11.013
+
+    .. versionadded:: 1.3.2
```

Comment:
this will be added in 1.4 unless it's backported (which I would not do as it's not a bug fix)

Reply:
```suggestion
    .. versionadded:: 1.4
```